options/ansi/generic/file-io.cpp

```diff
@@ -221,10 +221,14 @@ int abstract_file::write(const char *buffer, size_t max_size, size_t *actual_siz
 	return 0;
 }
 
-void abstract_file::unget(char c) {
+int abstract_file::unget(char c) {
+	if(!__offset) {
+		return EOF;
+	}
 	__ensure(__offset);
 	__offset--;
 	__buffer_ptr[__offset] = c;
+	return 0;
```

Comment:
why not return ``c`` here and remove the entire block of changes below in favor of just return unget

Reply:
Done as requested, also updated the commit message as requested.